### PR TITLE
feat: add farcaster resolver

### PR DESCRIPTION
**Pull request by SamuelLHuber**
resolving https://github.com/snapshot-labs/stamp/issues/180 with changes from @Mr94t3z and adding in axios default params from utils

### CODE REVIEW

**Comment by bonustrack** [comment]
I think this API is not live updated, doesnt work for my adress for example: https://searchcaster.xyz/api/profiles?connected_address=0xeF8305E140ac520225DAf050e2f71d5fBcC543e7

**Comment by bonustrack** [comment]
- Use Neynar API instead of Searchcaster (seem like Searchcaster is missing data or is not live updated)
- Add the Farcaster resolver in the index file here: https://github.com/snapshot-labs/stamp/blob/master/src/resolvers/index.ts - Add a test here: https://github.com/snapshot-labs/stamp/tree/master/test/integration/resolvers

Also check here our codestyle: https://github.com/snapshot-labs/guidebook/wiki/Codestyle

**Comment by SamuelLHuber** [comment]
You'll need to use the NEYNAR_API_KEY .env variable to test this yourself which you can get at https://dev.neynar.com/ @bonustrack

File: test/integration/resolvers/lens.test.ts [hunk 1]
```diff
@@ -8,15 +8,8 @@ describe('resolvers', () => {
...
-    it('should resolve with handle', async () => {
-      const result = await resolvers.lens('fabien.lens');
-
-      expect(result).toBeInstanceOf(Buffer);
-      expect(result.length).toBeGreaterThan(1000);
-    });
-
     it('should resolve with address', async () => {
-      const result = await resolvers.lens('0xeF8305E140ac520225DAf050e2f71d5fBcC543e7');
+      const result = await resolvers.farcaster('0x09CEdb7bb69f9F6DF646dBa107D2bAACda93D6C9')
```
Comment (bonustrack) [inline test/integration/resolvers/lens.test.ts:12]: Can you remove change on this file?

Comment (SamuelLHuber) [reply test/integration/resolvers/lens.test.ts]: describe('resolvers', () => {
  describe('lens', () => {
    it('should return false if missing', async () => {
      const result = await resolvers.lens('0x556B14CbdA79A36dC33FcD461a04A5BCb5dC2A70');

      expect(result).toBe(false);
    });

    it('should resolve with handle', async () => {
      const result = await resolvers.lens('fabien.lens');

      expect(result).toBeInstanceOf(Buffer);
      expect(result.length).toBeGreaterThan(1000);
    });

    it('should resolve with address', async () => {
      const result = await resolvers.lens('0xeF8305E140ac520225DAf050e2f71d5fBcC543e7');

      expect(result).toBeInstanceOf(Buffer);
      expect(result.length).toBeGreaterThan(1000);
    });
  });
});

Comment (SamuelLHuber) [reply test/integration/resolvers/lens.test.ts]: fixed the copy & paste overwrite, sorry

File: src/resolvers/farcaster.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import axios from 'axios';
+import { getAddress, isAddress } from '@ethersproject/address';
+import { resize } from '../utils';
+import { max } from '../constants.json';
+import { fetchHttpImage, axiosDefaultParams } from './utils';
```
Comment (bonustrack) [inline src/resolvers/farcaster.ts:5]: Can you remove this, this is not used, also please run the lint script
```suggestion
import { fetchHttpImage } from './utils';
```

**Comment by bonustrack** [comment]
When I try I got this error: "Error resolving Farcaster avatar: Error: Provided config url is not valid"

File: src/resolvers/farcaster.ts (hunk 2 above)
Comment (SamuelLHuber) [reply src/resolvers/farcaster.ts:5]: @bonustrack can't run the lint script as yarn fails with issues in regards to module 'canvas' node-gyp errors

**Comment by Mr94t3z** [comment]
ref #200

File: src/resolvers/farcaster.ts (hunk 2 above)
Comment (SamuelLHuber) [reply src/resolvers/farcaster.ts]: ran eslint manually to get linting

**Comment by SamuelLHuber** [comment]
> ref #200

if you want to take this feel free to, my yarn is running into issues with the canvas module so I am unable to execute anything right now.

**Comment by wa0x6e** [comment]
Closed via #201

**Pull request closed without merging**